superset/assets/src/dashboard/actions/dashboardState.js

```diff
@@ -261,6 +261,18 @@ export function removeSliceFromDashboard(id) {
   };
 }
 
+export const SET_COLOR_SCHEME = 'SET_COLOR_SCHEME';
+export function setColorScheme(colorScheme) {
+  return { type: SET_COLOR_SCHEME, colorScheme };
+}
+
+export function setColorSchemeAndUnsavedChanges(colorScheme) {
+  return dispatch => {
+    dispatch(setColorScheme(colorScheme));
+    dispatch(setUnsavedChanges(true));
```

Comment:
setUnsavedChanges (and `hasUnsavedChanges` state) is shared with dashboard's layout change. If you change color scheme,  change dashboard layout, then undo a all dashboard layout change, the hasUnsavedChanges flag will be reset. 
you probably have to add some logic to make sure this flag carries changes from both.

Reply:
I added a bool flag to track unsaved color changes. I don't love the approach but since we want the layout stuff to be refactored to allow non-layout changes to be undo-able, this was the least amount of changes until that refactoring gets done.